### PR TITLE
feat: implement the call-to-action in the API creation and deployment

### PR DESCRIPTION
**Pull request by aelamrani**
## Issue

https://gravitee.atlassian.net/browse/APIM-2003

## Description

Added the CTA as described in the mockups to fit the basic AC:

- CE user sees the license banner on all listed pages.
- EE user does not see the license banner on all listed pages.

## Additional context

- UTM tags will be managed by https://gravitee.atlassian.net/browse/APIM-1994.
- There is currently no check on the API definition to validate if the API contains at least 1 undeployed EE plugin so we can show the banner only when it is appropriate. Same for the summary creation of an API.
- There is also no check on the plugins deployed to know if we display the banner or not when selecting entrypoint option (even if the user has no EE entrypoint plugin in his license, he won't see the banner => we just check if he has a license).

Capture d’écran 2023-07-05 à 16 04 41
Capture d’écran 2023-07-05 à 16 05 25
Capture d’écran 2023-07-05 à 16 05 05
Capture d’écran 2023-07-05 à 16 04 51

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-mggehrzxrr.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.scss [hunk 1]
```diff
@@ -37,3 +37,7 @@ $typography: map.get(gio.$mat-theme, typography);
     margin: 0;
   }
 }
+
+gio-banner-error {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.scss]: What do you think about adding a dedicated class to add this style? like `license__banner`
We're trying to respect "css bem"

File: gravitee-apim-console-webui/src/components/gio-ee-unlock-dialog/gio-ee-unlock-dialog.component.html [hunk 2]
```diff
@@ -23,13 +23,30 @@
...
   <div class="gio-ee-unlock-dialog__content__action">
-    <button class="gio-ee-unlock-dialog__content__action__button" mat-button mat-stroked-button color="accent">
+    <a
+      class="gio-ee-unlock-dialog__content__action__button"
+      mat-button
+      mat-stroked-button
+      color="accent"
+      target="_blank"
+      [href]="
+        'https://www.gravitee.io/contact-us-alerts?utm_source=oss_apim&utm_campaign=oss_apim_to_ee_apim&utm_medium=' +
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/components/gio-ee-unlock-dialog/gio-ee-unlock-dialog.component.html]: Shouldn't you use the link from GioLicenseService > ctaLinkPrefixa?

File: gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.service.ts [hunk 3]
```diff
@@ -16,67 +16,96 @@
 import { HttpClient } from '@angular/common/http';
 import { Inject, Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
-import { map, shareReplay } from 'rxjs/operators';
+import { map, shareReplay, tap } from 'rxjs/operators';
+import { action } from '@storybook/addon-actions';
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.service.ts]: You can't have that dependency here. It's for storybook and not for the APIM console. 
maybe a bad copy and paste ?

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.service.ts]: I've just seen this used elsewhere - you can simply delete the action and parent pipe. This dialog returns no value to be interpreted this is not necessary.

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.html [hunk 4]
```diff
@@ -81,6 +81,18 @@
...
+            This configuration requires an enterprise license
+            <span gioBannerBody
+              >Request a license to unlock enterprise functionality, such as support for event brokers, asynchronous APIS, and Webhook
+              subscriptions.</span
+            >
+            <div gioBannerAction>
+              <button mat-raised-button color="basic" (click)="licenseService.displayUpgradeCta($event)">Request upgrade</button>
+            </div>
+          </gio-banner-error>
+        </div>
```
Comment (a-cordier) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-2-entrypoints/step-2-entrypoints-0-architecture.component.html]:
```suggestion
<gio-banner type="error" icon="gio:universe">
            This configuration requires an enterprise license
            <span gioBannerBody
              >Request a license to unlock enterprise functionality, such as support for event brokers, asynchronous APIS, and Webhook
              subscriptions.</span
            >
            <div gioBannerAction>
              <button mat-raised-button color="basic" (click)="licenseService.displayUpgradeCta($event)">Request upgrade</button>
            </div>
          </gio-banner>
```

This way you will get the same icon as in the mockup

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-5-summary/step-5-summary.component.html [hunk 5]
```diff
@@ -125,6 +125,18 @@
...
+      <gio-banner-error>
+        This configuration requires an enterprise license
+        <span gioBannerBody
+          >Request a license to unlock enterprise functionality, such as support for event brokers, asynchronous APIS, and Webhook
+          subscriptions.</span
+        >
+        <div gioBannerAction>
+          <button mat-raised-button color="basic" (click)="licenseService.displayUpgradeCta($event)">Request upgrade</button>
+        </div>
+      </gio-banner-error>
```
Comment (a-cordier) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-5-summary/step-5-summary.component.html]:
```suggestion
      <gio-banner type="error" icon="gio:universe">
        This configuration requires an enterprise license
        <span gioBannerBody
          >Request a license to unlock enterprise functionality, such as support for event brokers, asynchronous APIS, and Webhook
          subscriptions.</span
        >
        <div gioBannerAction>
          <button mat-raised-button color="basic" (click)="licenseService.displayUpgradeCta($event)">Request upgrade</button>
        </div>
      </gio-banner>
```

File: gravitee-apim-console-webui/src/management/api/portal/details/api-portal-details-danger-zone/api-portal-details-danger-zone.component.html [hunk 6]
```diff
@@ -35,6 +35,18 @@ <h3 class="danger-card__title">Danger Zone</h3>
...
+      <div *ngIf="!hasLicense">
+        <gio-banner-error>
+          This action is only available for users with an enterprise license
+          <span gioBannerBody
+            >Request a license to upgrade your account and unlock publications of your API on the enterprise gateway.</span
+          >
+          <div gioBannerAction>
+            <button mat-raised-button color="basic" (click)="licenseService.displayUpgradeCta($event)">Request upgrade</button>
+          </div>
+        </gio-banner-error>
```
Comment (a-cordier) [inline gravitee-apim-console-webui/src/management/api/portal/details/api-portal-details-danger-zone/api-portal-details-danger-zone.component.html]:
```suggestion
       <gio-banner type="error" icon="gio:universe">
          This action is only available for users with an enterprise license
          <span gioBannerBody
            >Request a license to upgrade your account and unlock publications of your API on the enterprise gateway.</span
          >
          <div gioBannerAction>
            <button mat-raised-button color="basic" (click)="licenseService.displayUpgradeCta($event)">Request upgrade</button>
          </div>
        </gio-banner>
```

**Review by a-cordier — CHANGES_REQUESTED** [review]

File: gravitee-apim-console-webui/src/components/gio-ee-unlock-dialog/gio-ee-unlock-dialog.component.html (hunk 2 above)
Comment (a-cordier) [reply gravitee-apim-console-webui/src/components/gio-ee-unlock-dialog/gio-ee-unlock-dialog.component.html]: There's another PR coming to handle that here https://github.com/gravitee-io/gravitee-api-management/pull/4552

**Review by a-cordier — APPROVED** [review]

File: gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.service.ts [hunk 7]
```diff
@@ -73,10 +88,17 @@ export class GioLicenseService {
...
 
+  public readonly ctaLinkPrefix: string =
+    'https://www.gravitee.io/contact-us-alerts?utm_source=oss_apim&utm_campaign=oss_apim_to_ee_apim&utm_medium=';
+
   getLicense() {
     return this.loadLicense$;
   }
 
+  hasLicense() {
+    return this.loadLicense$.toPromise().then((license) => !!license.features.length);
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.service.ts:99]: Ha, that's not the way with Angular 2+. 
Here you need to keep the observable and add an ` *ngIf="!hasLicense$ | async"` in the html 
and add un controller : 
`private hasLicense$ = this.licenseService.hasLicense()`

**Pull request merged**